core-rust/state-manager/src/transaction/ledger_transaction.rs

```diff
@@ -130,9 +130,33 @@ const GENESIS_TRANSACTION_SYSTEM_TRANSACTION_DISCRIMINATOR: u8 = 1;
 
 #[derive(Debug, Clone, PartialEq, Eq, ManifestCategorize, ManifestEncode, ManifestDecode)]
 pub struct FlashTransactionV1 {
+    pub nonce: u64,
     pub state_updates: StateUpdates,
 }
 
+impl FlashTransactionV1 {
+    pub fn prepare(&self) -> Result<PreparedFlashTransactionV1, PrepareError> {
+        let state_updates_hash = hash(scrypto_encode(&self.state_updates).unwrap());
+        let flash_hash = HashAccumulator::new()
+            .update([
+                TRANSACTION_HASHABLE_PAYLOAD_PREFIX,
+                TransactionDiscriminator::V1Flash as u8,
+            ])
+            .update(state_updates_hash)
+            .update(self.nonce.to_le_bytes().as_slice())
```

Comment:
It's weird to serialize for storage as little endian....

And actually I'd quite strongly rather we named the FlashTransaction rather than gave it a nonce. Would that be okay?

The name can then be used to describe the content of the transaction. Eg `anenome-pool-code-update` or whatever.

I guess we'd also need to adjust the Core API models.

Reply:
(although if we want to add a nonce of the state version too to be extra sure (as well as the name) we can?) - if we're worried the commit step wouldn't detect the duplicate hash?

Reply:
I'm okay with named transactions, will change.

Commit should detect duplicates :+1:

Reply:
(I'll address this in: https://github.com/radixdlt/babylon-node/pull/803)